tide/master/rest/RestWindows.h

```diff
@@ -0,0 +1,84 @@
+/*   1. Redistributions of source code must retain the above         */
+/*      copyright notice, this list of conditions and the following  */
+/*      disclaimer.                                                  */
+/*                                                                   */
+/*   2. Redistributions in binary form must reproduce the above      */
+/*      copyright notice, this list of conditions and the following  */
+/*      disclaimer in the documentation and/or other materials       */
+/*      provided with the distribution.                              */
+/*                                                                   */
+/*    THIS  SOFTWARE IS PROVIDED  BY THE  UNIVERSITY OF  TEXAS AT    */
+/*    AUSTIN  ``AS IS''  AND ANY  EXPRESS OR  IMPLIED WARRANTIES,    */
+/*    INCLUDING, BUT  NOT LIMITED  TO, THE IMPLIED  WARRANTIES OF    */
+/*    MERCHANTABILITY  AND FITNESS FOR  A PARTICULAR  PURPOSE ARE    */
+/*    DISCLAIMED.  IN  NO EVENT SHALL THE UNIVERSITY  OF TEXAS AT    */
+/*    AUSTIN OR CONTRIBUTORS BE  LIABLE FOR ANY DIRECT, INDIRECT,    */
+/*    INCIDENTAL,  SPECIAL, EXEMPLARY,  OR  CONSEQUENTIAL DAMAGES    */
+/*    (INCLUDING, BUT  NOT LIMITED TO,  PROCUREMENT OF SUBSTITUTE    */
+/*    GOODS  OR  SERVICES; LOSS  OF  USE,  DATA,  OR PROFITS;  OR    */
+/*    BUSINESS INTERRUPTION) HOWEVER CAUSED  AND ON ANY THEORY OF    */
+/*    LIABILITY, WHETHER  IN CONTRACT, STRICT  LIABILITY, OR TORT    */
+/*    (INCLUDING NEGLIGENCE OR OTHERWISE)  ARISING IN ANY WAY OUT    */
+/*    OF  THE  USE OF  THIS  SOFTWARE,  EVEN  IF ADVISED  OF  THE    */
+/*    POSSIBILITY OF SUCH DAMAGE.                                    */
+/*                                                                   */
+/* The views and conclusions contained in the software and           */
+/* documentation are those of the authors and should not be          */
+/* interpreted as representing official policies, either expressed   */
+/* or implied, of Ecole polytechnique federale de Lausanne.          */
+/*********************************************************************/
+
+#ifndef RESTWINDOWS_H
+#define RESTWINDOWS_H
+
+#include <servus/serializable.h> // base class
+
+#include "scene/ContentWindow.h"
+#include "scene/DisplayGroup.h"
+#include "MasterConfiguration.h"
+#include <QObject>
+#include "RestServer.h"
+
+/**
+ * Exposes display group content to ZeroEQ http server.
+ */
+class RestWindows : public QObject, public servus::Serializable
+{
+    Q_OBJECT
+
+public:
+    /**
+     * Construct dynamic content used by REST interface.
+     *
+     * @param config MasterConfiguration object which is exposed.
+     */
+    RestWindows( RestServer& server, const MasterConfiguration& config,
+                 const DisplayGroup& displayGroup );
+
+    /** @return the string used as an endpoint by REST interface. */
+    std::string getTypeName() const final;
+
+public slots:
+
```

Comment:
extra \n, the slots should be private functions it seems (else they must be documented)